### ISSUE TITLE
twistd script show unwanted messagge

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4030]
|<img alt="marco.giusti@...'s avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| marco.giusti@... reported|
|-|-|
|Trac ID|trac#4030|
|Type|defect|
|Created|2009-09-16 19:40:05Z|
|Branch|https://github.com/twisted/twisted/tree/scripts-options-longdesc-4030|

here the output:

```
$ twistd --help
...
  -b, --debug            run the application in the Python Debugger (implies
                         nodaemon), sending SIGUSR2 will drop into debugger
  -r, --reactor=         Which reactor to use (see --help-reactors for a list of
                         possibilities)
      --help             Display this help and exit.

module(name[, doc])

Create a module object. The name must be a string; the optional doc argument can
have any type.
Commands:
    ftp                 An FTP server.
    telnet              A simple, telnet-based remote debugging service.
    socks               A SOCKSv4 proxy service.
    manhole-old         An interactive remote debugger service.
...
```

this happens just if twisted is installed using setuptools: in this case _twistd_ script is just a wrapper to the real script (_Twisted-.....egg/EGG-INFO/scripts/twistd_) and the global namespace is replaced (!__main!__ and !__main!__.!__doc!__ too).

adding a docstring to twisted/scripts/twistd.py is enough.

<details><summary>Searchable metadata</summary>

```
trac-id__4030 4030
type__defect defect
reporter__marco_giusti____ marco.giusti@...
priority__normal normal
milestone__Twisted_9_0 Twisted-9.0
branch__branches_scripts_options_longdesc_4030 branches/scripts-options-longdesc-4030
branch_author__therve therve
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1253130005000000 1253130005000000
changetime__1258758524000000 1258758524000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)|@therve set owner to @therve|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_2"></a>|@therve commented|
|-|-|

(In [27585]) Branching to 'scripts-options-longdesc-4030'

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_3"></a>|@therve commented|
|-|-|

(In [27586]) Add longdesc to script options, to workaround a stupid setuptools bug mangling
module documentation, and for friendliness

Refs [#4030](https://github.com/twisted/twisted/issues/4030)

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)|@therve removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)<a name="note_5"></a>|@radix set owner to @therve|
|-|-|

1. I think the descriptions you wrote could be improved by following this format:

"<command> <verb>s <thing>s"

or

"<command> is a <foo> that <bar>s".

It'd be a minor improvement, but one that might mean the difference between confusion and understanding for a few users.

Like:

+    longdesc = ("Loads and executes a suite of unit tests, obtained from "
+                "modules, packages and files listed on the command line.")

could be

"trial loads and executes a suite of unit tests,..."

and

+    longdesc = ("SSHv2 client for logging into a remote machine and "
+                "executing commands.")

could be

"conch is a SSHv2 client that allows logging into ..."




I guess that's it. Thanks, +1.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_6"></a>|@therve set status to `closed`|
|-|-|

(In [27591]) Merge scripts-options-longdesc-4030

Author: therve
Reviewer: radix
Fixes [#4030](https://github.com/twisted/twisted/issues/4030)

Add longdesc to script descriptions, to workaround a bug in setuptools which
mangles __main__.__doc__.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
